dev-tools/globals.ts

```diff
@@ -720,17 +720,17 @@ interface RuleTable extends Map<string, string> {
 	getReason: (key: string) => string
 }
 
-interface Format extends BasicEffect, FormatsData {
-	effectType: 'Format' | 'Ruleset' | 'Rule' | 'ValidatorRule'
-	baseRuleset: string[]
-	banlist: string[]
-	customRules: string[] | null
-	defaultLevel: number
-	maxLevel: number
-	noLog: boolean
-	ruleset: string[]
+interface Format extends Readonly<BasicEffect & FormatsData> {
```

Comment:
Can these all be removed entirely now that they're redundant with the class definitions in `dex-data`?

Reply:
Just tried removing them in `globals.ts` = causes things to fail.

However, even if we could remove them from here,  I'd want them in both places as documentation/for people who aren't intimately familiar with how Typescript's merging works. IMO, ideally we'd only have definitions in one place (ie. nix globals and instead [import types](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#import-types)), but if we need to have the definition in two places (`dex-data` and `globals`) its a lot less confusing if they match.